DEVELOPING.md

```diff
@@ -0,0 +1,126 @@
+# Logstash Development Workflow
+
+This explains how to develop in the logstash core code-base. We will use to following fictional directories:
+
+- `/logstash/` is the dir containing your local fork clone of logstash
+- `/plugins/...` is the dir containing your local fork clones of logstash plugins or any other local private plugins
+
+## Setup
+
+Logstash uses [JRuby](http://jruby.org/) which gets embedded in the `vendor/jruby/` directory. It is **highly recommended** but not mandatory that you also use JRuby as your local Ruby interpreter and for this you should consider using a Ruby version manager such as [RVM](https://rvm.io/) or [rbenv](https://github.com/sstephenson/rbenv). It is possible to run the rake tasks and the `bin/` commands without having JRuby locally installed in which case the embedded JRuby will be used automatically.
+
+You will also need to have a JVM installed, we recommend using Java 1.8.
+
+`Rake` and `Bundler` should also be installed in your environment.
```

Comment:
I think we should highlight this a bit more! we got issues from time to time that end up being because people don't have them installed beforehand. Also is important to notice the need of rubygems version > 2.5.x

see 

```
If you are seeing errors that look like

$ rake bootstrap
Installing minitar >= 0 because the build process needs it.
[bootstrap] Fetching and installing gem: minitar (>= 0)
rake aborted!
LoadError: no such file to load -- archive/tar/minitar
/Users/<user>/projects/logstash/rakelib/vendor.rake:17:in `untar'
/Users/<user>/projects/logstash/rakelib/vendor.rake:86:in `(root)'
Tasks: TOP => bootstrap => vendor:all => vendor:jruby
(See full trace by running task with --trace)
then you may need to update your version of rubygems. Run gem -v to see the version of rubygems installed. Version 2.5.2 or higher should work. To update rubygems run gem update --system (you may need to run with sudo if you're using your system Ruby environment).
```

from current readme file.